court/src/routes/auth0.rs

```diff
@@ -0,0 +1,302 @@
+use std::env::VarError; // TODO remove
+
+use rocket::http::{Cookie, CookieJar, SameSite, Status};
+use rocket::response;
+use rocket::serde::{Deserialize, Serialize};
+use rocket::State;
+
+use crypto_hash::hex_digest;
+use jsonwebtoken::{
+    decode, decode_header, jwk, jwk::AlgorithmParameters, Algorithm, DecodingKey, Validation,
+};
+use rand::Rng;
+
+use crate::models::{NewSession, NewUser, Session};
+use crate::services::{auth0, establish_connection, users};
+use crate::{errors, services};
+
+pub fn random_state_string() -> String {
+    use rand::{distributions::Alphanumeric, thread_rng};
+    let mut rng = thread_rng();
+    let chars: String = (0..7).map(|_| rng.sample(Alphanumeric) as char).collect();
```

Comment:
```suggestion
    let chars: String = (0..16).map(|_| rng.sample(Alphanumeric) as char).collect();
```
Test that this works. Seems best to not have it quite so short